<docs>
	<doc for="NullReferenceException.NullReferenceException3">
		<summary>
			<para>Initializes a new instance of the <see cref="System.NullReferenceException"/> class with serialized data.</para>
		</summary>
		<param name="info">The object that holds the serialized object data.</param>
		<param name="context">The contextual information about the source or destination.</param>
		<remarks>
			<para>This constructor is called during
      deserialization to reconstitute the exception object transmitted over a stream.
      For more information, see <see topic="cpconSerialization"/>.</para>
		</remarks>
		<seealso topic="cpconSerialization"/>
	</doc><doc for="NullReferenceException.NullReferenceException">
		<overload>
			<para>Initializes a new instance of the <see cref="System.NullReferenceException"/> class.</para>
		</overload>
		<summary>
   Initializes a new instance of the <see cref="System.NullReferenceException"/> class, setting the <see cref="System.Exception.Message"/>
   property of the new instance to a system-supplied message that describes the
   error, such as "The value 'null' was found where an instance of an object was
   required." This message takes into account the current system culture.
</summary>
		<remarks>
			<para>The following table shows the initial property values for an instance of
   <see cref="System.NullReferenceException"/>.</para>
			<list type="table"><listheader>
					<term> Property</term>
					<description>Value</description>
				</listheader>
				<item>
					<term><see cref="System.Exception.InnerException"/></term>
					<description><see langword="null"/>.</description>
				</item>
				<item>
					<term><see cref="System.Exception.Message"/></term>
					<description>The localized error message string.</description>
				</item>
			</list>
		</remarks>
	</doc><doc for="NullReferenceException.NullReferenceException1">
		<summary>
			<para>Initializes a new instance of the <see cref="System.NullReferenceException"/> class with
   a specified error message.</para>
		</summary>
		<param name="message">A <see cref="System.String"/> that describes the error. The content of <paramref name="message"/> is intended to be understood by humans. The caller of this constructor is required to ensure that this string has been localized for the current system culture.</param>
		<remarks>
			<para>The following table shows the initial property values for an instance of
   <see cref="System.NullReferenceException"/>.</para>
			<list type="table"><listheader>
					<term> Property</term>
					<description>Value</description>
				</listheader>
				<item>
					<term><see cref="System.Exception.InnerException"/></term>
					<description><see langword="null"/>.</description>
				</item>
				<item>
					<term><see cref="System.Exception.Message"/></term>
					<description>The error message string.</description>
				</item>
			</list>
		</remarks>
	</doc><doc for="NullReferenceException.NullReferenceException2">
		<summary>
			<para>Initializes a new instance of the <see cref="System.NullReferenceException"/> 
class with a specified error message and a reference to the
inner exception that is the cause of this exception.</para>
		</summary>
		<param name="message">The error message that explains the reason for the exception.</param>
		<param name="innerException">The exception that is the cause of the current exception. If the <paramref name="innerException"/> parameter is not <see langword="null"/>, the current exception is raised in a <see langword="catch"/> block that handles the inner exception.</param>
		<remarks>
			<para>An exception that is thrown as a direct result of a 
      previous exception should include a reference to the previous exception in
      the <see cref="System.Exception.InnerException"/> property. The
   <see cref="System.Exception.InnerException"/> 
   property returns the same value that is passed into the constructor, or
<see langword="null"/> if the <see cref="System.Exception.InnerException"/> 
property does not supply the inner exception value to
the constructor. </para>
			<para>The following table shows the initial property values for an instance of
<see cref="System.NullReferenceException"/>.</para>
			<list type="table"><listheader>
					<term>Property</term>
					<description>Value</description>
				</listheader>
				<item>
					<term><see cref="System.Exception.InnerException"/></term>
					<description>The inner exception reference.</description>
				</item>
				<item>
					<term><see cref="System.Exception.Message"/></term>
					<description>The error message string.</description>
				</item>
			</list>
		</remarks>
		<seealso cref="System.Exception"/>
		<seealso topic="cpconHandlingThrowingExceptions"/>
	</doc><doc for="NullReferenceException">
		<summary>
			<para> The exception that is thrown when there is an attempt
      to dereference a null object reference.</para>
		</summary>
		<remarks>
			<para>Note that applications throw the <see cref="System.ArgumentNullException"/>
exception rather than the <see cref="System.NullReferenceException"/> exception
discussed here. </para>
			<para>The following Microsoft intermediate language (MSIL) instructions throw <see cref="System.NullReferenceException"/>:</para>
			<list type="bullet"><item>
					<term>
      
      callvirt</term>
				</item>
				<item>
					<term>
      
      cpblk</term>
				</item>
				<item>
					<term>
      
      cpobj</term>
				</item>
				<item>
					<term>
      
      initblk</term>
				</item>
				<item>
					<term>
      
      ldelem.&lt;type&gt;</term>
				</item>
				<item>
					<term>
      
      ldelema</term>
				</item>
				<item>
					<term>
      
      ldfld</term>
				</item>
				<item>
					<term>
      
      ldflda</term>
				</item>
				<item>
					<term>
      
      ldind.&lt;type&gt;</term>
				</item>
				<item>
					<term>
      
      ldlen</term>
				</item>
				<item>
					<term>
      
      stelem.&lt;type&gt;</term>
				</item>
				<item>
					<term>
      
      stfld</term>
				</item>
				<item>
					<term>
      
      stind.&lt;type&gt;</term>
				</item>
				<item>
					<term>
      
      throw</term>
				</item>
				<item>
					<term>
      
      unbox</term>
				</item>
			</list>
			<para><see cref="System.NullReferenceException"/> uses the HRESULT
COR_E_NULLREFERENCE, which has the value 0x80004003.</para>
			<para>For a list of initial property values for an instance of <see cref="System.NullReferenceException"/> , 
see the <see cref="M:System.NullReferenceException.#ctor"/> constructors.</para>
		</remarks>
		<seealso cref="System.Exception"/>
		<seealso topic="cpconHandlingThrowingExceptions"/>
	</doc>
</docs>
